docs/user/exercises/programming-exercise-AI-proof-guidelines.inc

```diff
@@ -0,0 +1,487 @@
+
+   Generated solution:
+
+   .. code:: java
+
+      public class Pumpkin {
+
+
+          private double weight;
+          private String face;
+
+          public Pumpkin(double weight) {
+          this.weight = weight;
+          this.face = "";
+
+
+          }
+
+
+      }
+
+   Example solution:
+
+   .. code:: java
+
+      public class Pumpkin {
+          private double weight;
+          private String face;
+          private String type;
+          private boolean seeds;
+
+          public Pumpkin(String type,
+                  double weight) {
+              this.type = type;
+              this.weight = weight;
+              face = "";
+              seeds = true;
+          }
+
+      }
+
+
+-  **Depictions of functionalities:** Rather than explaining how a
+   concept works in theory, provide depictions of the functionality.
+   This encourages students to understand the basic code instead of
+   going to AI for answers. Examples that were used in previous
+   exercises:
+
+
+   .. figure:: programming/AI-proof-exercises/List-generics.png
+      :align: center
+      :alt: This figure showcases the functionality of a linked-list
+      :width: 50%
+
+
+      Depiction shows the functionality of a linked list with 3 integer values
+      as a basic concept for understanding linked lists. The task of this exercise
+      were to implement the common List functionalities from scratch rather than
+      rely on the standard methods from java. ListNode and LinkedList were
+      therefore own classes that needed to be filled with the required
+      functionalities.
+
+
+   This depiction provides an understanding of how a linked list should look
+   like after the add function was called.
+
+   .. figure:: programming/AI-proof-exercises/add.png
+      :align: center
+      :width: 50%
+
+
+      Figure shows what a list should look like after the add functionality was called. Picture from the Problem statement
+
+-  **Java frameworks:** Using Java frameworks, such as JavaFX, is challenging for AI
+   and helps to visualize projects in a more interactive way
+   for students through gamification. How can we incorporate that for
+   our exercises?
+
+   -  Provide Graphical User Interfaces (GUI) for
+      students with which they can also test the functionality of their
+      code.
+
+   -  Give exercises that use GUIs because AI has difficulty
+      solving them accurately.
+
+   -  Provide pictures of how the GUI should look in the end.
+
+      Students needed to implement the GUI based on the this figure that
+      was given in the problem statement.
+
+      .. figure:: programming/AI-proof-exercises/GUI-upgrade-for-game.png
+         :align: center
+         :width: 30%
+
+
+         Example GUI from homework exercise
+
+   -  Providing GIFs is also possible and shows a good way to show the dynamic behaviour of the GUI.
+      This also ensures less likelyhood to copy this format into various AI tools.
```

Comment:
This format also ensures a lower likelihood of students copying it into various AI tools.